ide/languages.hcl/src/org/netbeans/modules/languages/hcl/HCLSemanticAnalyzer.java

```diff
@@ -87,25 +85,33 @@ public void cancel() {
         cancelled = true;
     }
     
-    protected abstract class Highlighter extends HCLElement.BAEVisitor {
-        protected final Map<OffsetRange, Set<ColoringAttributes>> work = new HashMap<>();
+    protected abstract class Highlighter {
+        protected final Map<OffsetRange, Set<ColoringAttributes>> work = new IdentityHashMap<>();
```

Comment:
is this due to the way how `OffsetRange` "calculates" its hash code?

Reply:
No it turned out that's not really needed. the `HashMap` -> `IdentityHashMap` change was required only in `SourceRef` the record generated hash codes seemed to be "too perfect", causing troubles in coloring.

Will revert back to `HashMap` here.